monitoring/v1alpha1/types.go

```diff
@@ -0,0 +1,280 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// NOTE: The AlertingRule type is a direct copy of the upstream PrometheusRule
+// type from prometheus-operator.  The only difference at the moment is that we
+// don't allow recording rules in OpenShift.  All rules must be alerting rules,
+// but outside of that restriction, each AlertingRule will result in a 1:1 alike
+// PrometheusRule object being created.
+//
+// See the upstream docs here:
+// - https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md
+
+// AlertingRule represents a set of user-defined Prometheus rule groups containing
```

Comment:
what do these do?  Why would I want to use them?

Reply:
I tried to expand this, and removed the note from the top of the file that wasn't part of the godoc. I tried to explain everything directly in the godoc for this type. Let me know if it needs something more.